code/modules/atmospherics/machinery/components/fusion/hfr_main_processes.dm

```diff
@@ -350,149 +208,177 @@
+		if(2)
+			if(moderator_list[/datum/gas/plasma] > 50)
+				internal_output.assert_gases(/datum/gas/bz)
+				internal_output.gases[/datum/gas/bz][MOLES] += scaled_production * 1.8
+				moderator_internal.gases[/datum/gas/plasma][MOLES] -= min(moderator_internal.gases[/datum/gas/plasma][MOLES], scaled_production * 1.75)
+			if(moderator_list[/datum/gas/proto_nitrate] > 20)
+				radiation *= 1.55
+				heat_output *= 1.025
+				internal_output.assert_gases(/datum/gas/stimulum)
+				internal_output.gases[/datum/gas/stimulum][MOLES] += scaled_production * 1.05
+				moderator_internal.gases[/datum/gas/proto_nitrate][MOLES] -= min(moderator_internal.gases[/datum/gas/proto_nitrate][MOLES], scaled_production * 1.35)
+		if(3, 4)
+			if(moderator_list[/datum/gas/plasma] > 10)
+				internal_output.assert_gases(/datum/gas/freon, /datum/gas/stimulum)
+				internal_output.gases[/datum/gas/freon][MOLES] += scaled_production * 0.15
+				internal_output.gases[/datum/gas/stimulum][MOLES] += scaled_production * 1.05
+				moderator_internal.gases[/datum/gas/plasma][MOLES] -= min(moderator_internal.gases[/datum/gas/plasma][MOLES], scaled_production * 0.45)
+			if(moderator_list[/datum/gas/freon] > 50)
+				heat_output *= 0.9
+				radiation *= 0.8
+			if(moderator_list[/datum/gas/proto_nitrate]> 15)
+				internal_output.assert_gases(/datum/gas/stimulum, /datum/gas/halon)
+				internal_output.gases[/datum/gas/stimulum][MOLES] += scaled_production * 1.25
+				internal_output.gases[/datum/gas/halon][MOLES] += scaled_production * 1.15
+				moderator_internal.gases[/datum/gas/proto_nitrate][MOLES] -= min(moderator_internal.gases[/datum/gas/proto_nitrate][MOLES], scaled_production * 1.55)
+				radiation *= 1.95
+				heat_output *= 1.25
+			if(moderator_list[/datum/gas/bz] > 100)
+				internal_output.assert_gases(/datum/gas/healium, /datum/gas/proto_nitrate)
+				internal_output.gases[/datum/gas/proto_nitrate][MOLES] += scaled_production * 1.5
+				internal_output.gases[/datum/gas/healium][MOLES] += scaled_production * 1.5
+				induce_hallucination(50 * power_level, delta_time)
+		if(5)
+			if(moderator_list[/datum/gas/plasma] > 15)
+				internal_output.assert_gases(/datum/gas/freon)
+				internal_output.gases[/datum/gas/freon][MOLES] += scaled_production *0.25
+				moderator_internal.gases[/datum/gas/plasma][MOLES] -= min(moderator_internal.gases[/datum/gas/plasma][MOLES], scaled_production * 1.45)
+			if(moderator_list[/datum/gas/freon] > 500)
+				heat_output *= 0.5
+				radiation *= 0.2
+			if(moderator_list[/datum/gas/proto_nitrate] > 50)
+				internal_output.assert_gases(/datum/gas/stimulum, /datum/gas/pluoxium)
+				internal_output.gases[/datum/gas/stimulum][MOLES] += scaled_production * 1.95
+				internal_output.gases[/datum/gas/pluoxium][MOLES] += scaled_production
+				moderator_internal.gases[/datum/gas/proto_nitrate][MOLES] -= min(moderator_internal.gases[/datum/gas/proto_nitrate][MOLES], scaled_production * 1.35)
+				radiation *= 1.95
```

Comment:
Would there be any point in making these defines?

Reply:
This and virtually every other scattered imperative block of coefficients is getting nuked from orbit in the content rework. Sneak peek, subject to change:

```dm
/datum/hfr_gas_effect
        var/datum/gas/gas
        /// Used to calculate instability.
        /// Instability is cyclic with the period bounded by the magnetic constrictor.
        /// Gases with no fusion power do not advance instability.
        var/fusion_power = 0
        /// Adjusts the base energy of the HFR.
        /// This is the amount of potential energy available to the reaction before conduction and radiation loss.
        /// Always in the range [0.25, 100]
        /// Should be easy to hit 100%, but should need to keep a good base of common gases.
        var/energy_multiplier = 20
        /// Modifies the power output of the HFR.
        /// Adjusts the core temperature and calculates radiation.
        /// Directly scales power output from the fuel, and therefore heat.
        /// Always in the range [0.25, 100]
        /// Should be fairly easy to hit 100%, but freon can tank the output if not compensated for.
        /// We achieve this by making the base value 60%, adding values over 100% for many common gases,
        /// then setting freon to an extremely low value to tank the log-average when present.
        var/power_multiplier = 60
        /// Modifies the heat output of the HFR.
        /// Directly scales derived energy and final heat output.
        /// Always in the range [0.25, 100]
        /// Should be easy to hit 100%, but requires plasma to get there.
        /// The default is set relatively low, so one must be sure the moderator mix is primarily plasma.
        /// Nitrogen, n2o, and freon can all heavily tank output.
        var/heat_multiplier = 35
        /// Modifies the amount of potential energy emitted as radiation.
        /// This reduces energy available to other effects, including production, but can be used to
        /// produce power.
        /// Always in the range [0.25, 100]
        /// Plasma keeps this down, as does nitrogen, so it's easy to suppress. Freon very heavily
        /// suppresses this.
        /// However, BZ and to a lesser extent proto-nitrate have very high modifiers, so you will
        /// need to be creative with energy use.
        var/radiation_emission_multiplier = 0.25
        /// Internal radiation emissions are partly reabsorbed by the reaction.
        /// Energy lost to emission, internal or external, reduces the energy available to the reaction.
        /// However, only externally emitted radiation becomes dangerous rads outside of the machine.
        /// Always in the range [0.25, 100]
        /// The default of 10% means a good proportion is reabsorbed, but not enough for the
        /// reactor to be safe around for long.
        var/radiation_absorption_multiplier = 10

        /// A tiered list of gases to produce when we're present.
        /// Either empty, or a list suitable for use with production_tier_by_fusion_level, below.
        /// It is expected that most gases do not produce other gases.
        var/production_list
        /// A mapping from fusion power level to the above gas tiers as output.
        /// The first gas in each list is considered the primary product.
        var/production_tier_by_fusion_level = list(
                list(1),
                list(2,1),
                list(2,3),
                list(3,2),
                list(4,3,2),
                list(5,6),
        )
        /// Gases not at the start of one of the above mapping lists will have their output multiplied by this much.
        var/production_secondary_rate          = 0.8
        /// When producing gases, we require this much of ourselves to produce efficiently.
        var/production_requirements_base       = 150
        /// As we approach level 3/4, we require this much less of ourselves.
        var/production_requirements_early_drop = 100
        /// As we enter level 5, we require this much more of ourselves.
        var/production_requirements_late_rise
        /// A multiplier on the production rate previously determined by heat output, fuel injection rate, and active recipes.
        var/production_inherent_rate           = 1
        /// A multiplier on the consumption rate previously determined by heat output, fuel injection rate, and active recipes.
        var/production_consumption_rate        = 1
        /// At fusion level 6, multiplies the production rate.
        var/production_overdrive_rate          = 3
```

It's hard to state just how nice it is to see everything that a gas does all together in one place.

`_multiplier` should probably be `_percent`, at least, but I'll get to that in time

Reply:
:)